### PR TITLE
switch GCR -> Artifact-Registry

### PR DESCRIPTION
**Pull request by ccwienk**
GCR has been deprecated [0] in favour of Artifact-Registry.

Thus, change push-targets for OCI-Images:

- europe-docker.pkg.dev/gardener-project/snapshots for snapshots
- europe-docker.pkg.dev/gardener-project/releases for releases
- europe-docker.pkg.dev/gardener-project/public for combined view of snapshots + releases

[0]
https://cloud.google.com/artifact-registry/docs/transition/transition-from-gcr

**Release note**:

```breaking operator
Change OCI Image Registry from GCR (`eu.gcr.io/gardener-project`) to Artifact-Registry (`europe-docker.pkg.dev/gardener-project/releases`). Users should update their references.

```

### CODE REVIEW

**Comment by gardener-robot-ci-3** [comment]
This PR proposes changes that would break the pipeline definition:
```
vpn2-gcr-to-ar: Traceback (most recent call last):
  File "/usr/lib/python3.11/site-packages/concourse/replicator.py", line 141, in render
    definition_descriptor = self._render(definition_descriptor)
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/concourse/replicator.py", line 225, in _render
    raise RuntimeError(
RuntimeError: An exception occured when rendering pipeline 'vpn2-gcr-to-ar': 

Traceback (most recent call last):
  File "/usr/lib/python3.11/site-packages/concourse/replicator.py", line 217, in _render
    definition_descriptor.pipeline = template.render(
  File "/usr/lib/python3.11/site-packages/mako/template.py", line 438, in render
    return runtime._render(self, self.callable_, args, data)
  File "/usr/lib/python3.11/site-packages/mako/runtime.py", line 874, in _render
    _render_context(
  File "/usr/lib/python3.11/site-packages/mako/runtime.py", line 916, in _render_context
    _exec_template(inherit, lclcontext, args=args, kwargs=kwargs)
  File "/usr/lib/python3.11/site-packages/mako/runtime.py", line 943, in _exec_template
    callable_(context, *args, **kwargs)
  File "memory:0x7f25f0cc0690", line 666, in render_body
    ${job(variant)}
  File "memory:0x7f25f0cc0690", line 0, in job
    
  File "memory:0x7f25f0cc0690", line 654, in render_job
    ${step(4, job_variant, job_variant.step(step_name))}
  File "memory:0x7f25f0cc0690", line 619, in step
    <%def name="job(job_variant)">
  File "memory:0x7f25f0cc0690", line 600, in render_step
    ${execute(2, job_step, job_variant, source_repo, source_repo_github_cfg_name)}
  File "memory:0x7f25f0cc0690", line 567, in execute
    <%def name="step(indent, job_variant, job_step)" filter="indent_func(indent),trim">
  File "memory:0x7f25f0cc0690", line 526, in render_execute
    ${component_descriptor_step(job_step=job_step, job_variant=job_variant, output_image_descriptors=image_descriptors_for_variant, indent=8)}
  File "/usr/lib/python3.11/site-packages/mako/template.py", line 438, in render
    return runtime._render(self, self.callable_, args, data)
  File "/usr/lib/python3.11/site-packages/mako/runtime.py", line 874, in _render
    _render_context(
  File "/usr/lib/python3.11/site-packages/mako/runtime.py", line 920, in _render_context
    _exec_template(callable_, context, args=args, kwargs=kwargs)
  File "/usr/lib/python3.11/site-packages/mako/runtime.py", line 943, in _exec_template
    callable_(context, *args, **kwargs)
  File "/usr/lib/python3.11/site-packages/concourse/steps/component_descriptor.mako", line 20, in render_component_descriptor_step
    ocm_repository_mappings = descriptor_trait.ocm_repository_mappings()
  File "/usr/lib/python3.11/site-packages/concourse/model/traits/component_descriptor.py", line 405, in ocm_repository_mappings
    repo_url = mapping['repository']
TypeError: string indices must be integers, not 'str'


```

**Comment by gardener-robot-ci-1** [comment]
This PR proposes changes that would break the pipeline definition:
```
<invalid YAML>: sequence entries are not allowed here
  in "<unicode string>", line 53, column 25:
                repository: - europe-docker.pkg.dev/gardener ... 
                            ^
```

**Comment by gardener-robot-ci-1** [comment]
The pipeline-definition has been fixed.

File: .ci/pipeline_definitions [hunk 1]
```diff
@@ -67,9 +79,3 @@ vpn2:
             internal_scp_workspace:
               channel_name: 'C9CEBQPGE' #sap-tech-gardener
               slack_cfg_name: 'scp_workspace'
-        component_descriptor:
-          component_name: 'github.com/gardener/vpn2'
```
Comment (ScheererJ) [inline .ci/pipeline_definitions:71]: Why did you remove the name? Is it no longer required or is it just set implicitly?

File: Makefile [hunk 2]
```diff
@@ -3,8 +3,8 @@
 # SPDX-License-Identifier: Apache-2.0
 
 VERSION                       := $(shell cat VERSION)
-REGISTRY                      := eu.gcr.io/gardener-project/gardener
-PREFIX                        := vpn
+REGISTRY                      := europe-docker.pkg.dev/gardener-project/public
+PREFIX                        := gardener/vpn
```
Comment (ScheererJ) [inline Makefile:7]: Please add `gardener` to `REGISTRY` and remove it from `PREFIX`.

**Review by ScheererJ — CHANGES_REQUESTED** [review]

File: .ci/pipeline_definitions (hunk 1 above)
Comment (ccwienk) [reply .ci/pipeline_definitions:71]: @ScheererJ : technically, component_name defaults to github-repository-url, so adding it will not have any effect (as long as the repository is not moved/renamed). If you prefer, I can add the `component_name` attribute again, of course.

File: Makefile (hunk 2 above)
Comment (ccwienk) [reply Makefile]: done

Comment (ScheererJ) [reply Makefile]: Thanks.

File: .ci/pipeline_definitions (hunk 1 above)
Comment (ScheererJ) [reply .ci/pipeline_definitions:71]: No, we can leave it as is for now.

**Review by ScheererJ — APPROVED** [review]
/lgtm
